### PR TITLE
Notch support

### PR DESCRIPTION
**Pull request by abelgardep**
BUGS & IMPROVEMENTS:

- [X] (1) App crashes on start [FIXED]

### CODE REVIEW

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 1]
```diff
@@ -133,6 +137,39 @@ protected void setupDrawer() {
                 mNavigationView.setItemIconTintList(null);
             }
 
+            //Notch support
+            mNavigationView.addOnAttachStateChangeListener(new View.OnAttachStateChangeListener() {
+                @Override
+                public void onViewAttachedToWindow(View v) {
+
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: I wouldn't put a white line in here

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 2]
```diff
@@ -133,6 +137,39 @@ protected void setupDrawer() {
...
+                    if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
+
+                        DisplayCutout displayCutout = v.getRootWindowInsets().getDisplayCutout();
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: Nor here

File: owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java [hunk 3]
```diff
@@ -133,6 +137,39 @@ protected void setupDrawer() {
...
+                        if (displayCutout != null) {
+                            RelativeLayout rl_drawer_active_user =
+                                    (RelativeLayout) findNavigationViewChildById(R.id.drawer_active_user);
+
+                            int orientation = getResources().getConfiguration().orientation;
+                            if (orientation == Configuration.ORIENTATION_PORTRAIT) {
+                                int displayCutoutDP = (displayCutout.getSafeInsetTop()) /
+                                                (getResources().getDisplayMetrics().densityDpi /
+                                                        DisplayMetrics.DENSITY_DEFAULT);
+                                rl_drawer_active_user.getLayoutParams().height =
```
Comment (davigonz) [inline owncloudApp/src/main/java/com/owncloud/android/ui/activity/DrawerActivity.java]: We normally use camelCase for variable names and _ for layout ids for example, as you can see in  `(RelativeLayout) findNavigationViewChildById(R.id.drawer_active_user);`

**Comment by davigonz** [comment]
@abelgardep some changes requested

**Comment by davigonz** [comment]
Code approved, ready to test

**Comment by jesmrec** [comment]
### (1)

Installing the branch from scratch, the app crashes. This is the stack trace: 

```
2019-03-18 12:04:13.180 11563-11563/? E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.owncloud.android.debug, PID: 11563
    java.lang.NoSuchMethodError: No virtual method getDisplayCutout()Landroid/view/DisplayCutout; in class Landroid/view/WindowInsets; or its super classes (declaration of 'android.view.WindowInsets' appears in /system/framework/framework.jar!classes2.dex)
        at com.owncloud.android.ui.activity.DrawerActivity$1.onViewAttachedToWindow(DrawerActivity.java:145)
        at android.view.View.dispatchAttachedToWindow(View.java:17456)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3326)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3333)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3333)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3333)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3333)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3333)
        at android.view.ViewGroup.dispatchAttachedToWindow(ViewGroup.java:3333)
        at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1668)
        at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1392)
        at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6752)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:911)
        at android.view.Choreographer.doCallbacks(Choreographer.java:723)
        at android.view.Choreographer.doFrame(Choreographer.java:658)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:897)
        at android.os.Handler.handleCallback(Handler.java:790)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6494)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
2019-03-18 12:04:13.186 766-776/? W/ActivityManager:   Force finishing activity com.owncloud.android.debug/com.owncloud.android.authentication.AuthenticatorActivity
2019-03-18 12:04:13.195 766-781/? I/ActivityManager: Showing crash dialog for package com.owncloud.android.debug u0
```

Android 8.x

**Comment by jesmrec** [comment]
Approved, now the notch does not collapse with the upper side of the drawer.

There are some conflicts in `DrawerActivity.java`, could you take  a look? then, after rebasing, we can merge

**Pull request merged**